app/scripts/controllers/permissions/selectors.js

```diff
@@ -40,47 +41,74 @@ export const getPermittedAccountsByOrigin = createSelector(
 );
 
 /**
- * Given the current and previous exposed accounts for each PermissionController
- * subject, returns a new map containing all accounts that have changed.
+ * Get the permitted chains for each subject, keyed by origin.
+ * The values of the returned map are immutable values from the
+ * PermissionController state.
+ *
+ * @returns {Map<string, string[]>} The current origin:chainIds[] map.
+ */
+export const getPermittedChainsByOrigin = createSelector(
+  getSubjects,
+  (subjects) => {
+    return Object.values(subjects).reduce((originToChainsMap, subject) => {
+      const caveats =
+        subject.permissions?.[PermissionNames.permittedChains]?.caveats || [];
+
+      const caveat = caveats.find(
+        ({ type }) => type === CaveatTypes.restrictNetworkSwitching,
+      );
+
+      if (caveat) {
+        originToChainsMap.set(subject.origin, caveat.value);
+      }
+      return originToChainsMap;
+    }, new Map());
+  },
+);
+
+/**
+ * Given the current and previous exposed origins for each PermissionController
+ * subject, returns a new map containing all origins that have changed.
  * The values of each map must be immutable values directly from the
  * PermissionController state, or an empty array instantiated in this
  * function.
  *
- * @param {Map<string, string[]>} newAccountsMap - The new origin:accounts[] map.
- * @param {Map<string, string[]>} [previousAccountsMap] - The previous origin:accounts[] map.
- * @returns {Map<string, string[]>} The origin:accounts[] map of changed accounts.
+ * @param {Map<string, string[]>} newOriginsMap - The new origin:string[] map.
+ * @param {Map<string, string[]>} [previousOriginsMap] - The previous origin:string[] map.
+ * @returns {Map<string, string[]>} The origin:string[] map of changed origins.
  */
-export const getChangedAccounts = (newAccountsMap, previousAccountsMap) => {
-  if (previousAccountsMap === undefined) {
-    return newAccountsMap;
+export const getChangedOrigins = (newOriginsMap, previousOriginsMap) => {
```

Comment:
I'm a bit confused by the use of the term `origins` here? Should it be `getChangedPermissions`? I'm not seeing how it relates to origins at all? Actually it really even more generic than `permissions` its sortof a pure agnostic comparison helper function no?

Reply:
`getChangedValues`?

Reply:
renamed this helper to `diffMap` and updated the internal variable names https://github.com/MetaMask/metamask-extension/pull/26635/commits/024597d47d18f6857ec23998e6b5124a41c130bb